File: dingtalk-mobile-client/src/services/task/TaskGroupService.js
import { stringify } from 'qs';
import request from './../../utils/HttpInvoker';

async function addTaskGroup(params) {
  return request(`/api/tm/task-group`, {
    method: 'POST',
    body : params
  });
}

async function editTaskGroup(params) {
  return request(`/api/tm/task-group/${params.id}`, {
    method: 'PUT',
    body: params,
  });
}

async function deleteTaskGroupById(params) {
  return request(`/api/tm/task-group/${params.id}`, {
    method: 'DELETE',
  });
}

async function findTaskGroupById(params) {
  return request(`/api/tm/task-group/${params.id}`, {
    method: 'GET',
  });
}

async function findTaskGroupList(params) {
  return request(`/api/tm/task-group/list?${stringify(params)}`, {
    method: 'GET',
  });
}

export default {
  addTaskGroup,
  editTaskGroup,
  deleteTaskGroupById,
  findTaskGroupById,
  findTaskGroupList
}

File: task-api/src/main/java/com/hozensoft/task/core/service/TaskGroupService.java
package com.hozensoft.task.core.service;

import com.hozensoft.struct.pagehelper.Limitable;
import com.hozensoft.system.utils.bean.ContextHolder;
import com.hozensoft.task.core.domain.TaskGroup;
import com.hozensoft.task.core.dto.TaskGroupDto;
import com.hozensoft.task.core.dto.TaskGroupItemDto;
import com.hozensoft.task.core.dto.TaskGroupQueryDto;
import com.hozensoft.task.core.dto.TaskGroupValueDto;

import java.util.List;

public interface TaskGroupService {

    TaskGroupDto addTaskGroup(ContextHolder holder, TaskGroupValueDto taskGroup);

    TaskGroupDto editTaskGroup(ContextHolder holder, TaskGroupValueDto taskGroup);

    void deleteTaskGroup(ContextHolder holder, String taskGroupId);

    TaskGroupDto findTaskGroupById(ContextHolder holder, String taskGroupId);

    TaskGroup loadTaskGroupById(ContextHolder holder, String taskGroupId);

    List<TaskGroupItemDto> findTaskGroupList(ContextHolder holder, TaskGroupQueryDto params, Limitable limitable);

}
